sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -192,6 +195,56 @@ def test_least_absolute_deviation():
     assert gbdt.score(X, y) > .9
 
 
+@pytest.mark.parametrize(
+    'y',
+    [([1., -2., 0.]),
+     ([0., 0., 0.]),
+     ])
+def test_poisson_y_positive(y):
+    # Test that ValueError is raised if either one y_i < 0 or sum(y_i) <= 0.
+    err_msg = r"loss='poisson' requires non-negative y and sum\(y\) > 0."
+    gbdt = HistGradientBoostingRegressor(loss='poisson', random_state=0)
+    with pytest.raises(ValueError, match=err_msg):
+        gbdt.fit(np.zeros(shape=(len(y), 1)), y)
+
+
+def test_poisson():
+    # For Poisson distributed target, Poisson loss should give better results
+    # than least squares measured in Poisson deviance as metric.
+    rng = np.random.RandomState(42)
+    n_train, n_test, n_features = 500, 100, 100
+    X = make_low_rank_matrix(n_samples=n_train+n_test, n_features=n_features,
+                             random_state=rng)
+    # We create a log-linear Poisson model and downscale coef as it will get
+    # exponentiated.
+    coef = rng.uniform(low=-2., high=+2., size=n_features) / np.amax(X, axis=0)
+    y = rng.poisson(lam=np.exp(X @ coef))
+    X_train, X_test = X[0:n_train, :], X[n_train:n_train+n_test, :]
+    y_train, y_test = y[0:n_train], y[n_train:n_train+n_test]
+    gbdt1 = HistGradientBoostingRegressor(loss='poisson', random_state=0)
```

Comment:
```suggestion
    gbdt1 = HistGradientBoostingRegressor(loss='poisson', random_state=rng)
```
For consistency. Though that won't matter: the RNG is only relevant for early stopping which won't be activated.